test/Interop/Cxx/operators/member-inline-module-interface.swift

```diff
@@ -15,3 +15,25 @@
 
 // CHECK: struct HasDeletedOperator {
 // CHECK: }
+
+// CHECK: struct ReadWriteIntArray {
+// CHECK:   subscript(x: Int32) -> Int32 { mutating get set }
```

Comment:
Yes, I think that would be best to solve as part of SR-12749. The C++ interop manifesto [mentions](https://github.com/apple/swift/blob/main/docs/CppInteroperabilityManifesto.md#const-qualified-member-functions-in-c-classes) that it would require some static analysis to make it both convenient & mostly correct, and I didn't want to dive into it in this patch.

For now I've tweaked the subscript generating logic slightly – we will now only add the `mutating` flag to an accessor if the underlying implementation (`__operatorSubscriptConst`) is mutable.

Reply:
Fantastic, thanks!